### PR TITLE
Feature/552 Behavior notification support

### PR DESCRIPTION
**Pull request by matthewwerny**
This is a prerequisite to multiple other tickets that are outstanding.  It provides the ability for one behavior to notify other behaviors on the same element that something of interest to the receiving behavior has occurred.

### CODE REVIEW

File: src/behavior/BehaviorInternalsImpl.ts [hunk 1]
```diff
@@ -33,6 +33,7 @@ import BehaviorFlags from "behavior/BehaviorFlags";
 
 const CHANNEL_NAME: string = "channelName";
 const MSG_NAME: string = "messageName";
+const EVENT_NAME: string = "cydran:behavior";
```
Comment (evilchia) [inline src/behavior/BehaviorInternalsImpl.ts:36]: Let's move this into HardValues so that it can be sued in Behavior.ts as well

**Review by evilchia — CHANGES_REQUESTED** [review]
See the comment about moving a text value into HardValues.ts

File: src/behavior/BehaviorInternalsImpl.ts (hunk 1 above)
Comment (matthewwerny) [reply src/behavior/BehaviorInternalsImpl.ts:36]: This value is exclusive to this class and never should be exposed for general use, so it isn't something that we want to advertise externally.

Comment (evilchia) [reply src/behavior/BehaviorInternalsImpl.ts:36]: There is another class that uses it there in your commit, otherwise I would agree with you.

Comment (matthewwerny) [reply src/behavior/BehaviorInternalsImpl.ts:36]: This is now isolated to this class alone.

**Review by evilchia — APPROVED** [review]

**Pull request merged**